### PR TITLE
Feat/metamask provider

### PR DESCRIPTION
**Pull request by Sednaoui**

### CODE REVIEW

File: crosscash-extension/package.json [hunk 1]
```diff
@@ -14,9 +15,8 @@
     "@types/react-dom": "^17.0.11",
     "loglevel": "^1.8.0",
     "react": "^17.0.2",
-    "react-browser-extension-scripts": "4.0.10",
     "react-dom": "^17.0.2",
-    "typescript": "^4.5.3",
+    "typescript": "^4.5.4",
```
Comment (Sednaoui) [inline crosscash-extension/package.json]: here

**Comment by QuocAnhVu** [comment]
You can open the test dApp and type `window.stream.write('hey');` in the console.

The button will also switch from "Click here to install metamask" to "Connect", signaling that the dApp detects the provider.

**Comment by Sednaoui** [comment]
> You can open the test dApp and type `window.stream.write('hey');` in the console.
> 
> The button will also switch from "Click here to install metamask" to "Connect", signaling that the dApp detects the provider.

I got 'hey' back 🥳

for some reason I never got the button to install metamask even without any extension installed on my browser. It was always connect.

**Comment by QuocAnhVu** [comment]
ready to merge

**Pull request merged**